components/raftstore/src/store/peer.rs

```diff
@@ -5636,7 +5637,7 @@ pub trait RequestInspector {
 
         // Local read should be performed, if and only if leader is in lease.
         // None for now.
-        match self.inspect_lease() {
+        match inspect_lease_res {
```

Comment:
See also: https://github.com/tikv/tikv/pull/14123#discussion_r1093084717

Reply:
Are you referring to the issue with lock processing? In my opinion, it's secure because we use snapshot readings and follow the correct snapshot read protocol for resolving locks.

In theory, it's also safe to use either the `local reader` or `read index` path if the peer considers itself a leader. For example:

```rust
if flags.contains(WriteBatchFlags::STALE_READ) && self.raft.is_follower() {
    return Ok(RequestPolicy::StaleRead);
}
// Use local read (leader) 
// or `read index` if follower or suspected leader for snapshot read.
// In these situations `safe_ts` is not needed to be considered.
```

Reply:
Stale read ts0 will ignore all locks because it's expected all locks will either be rollback or committed using a timestamp ts1 > ts0. But if a leader applies entries slowly, then an entry containing commit ts ts2 < ts0 can be in queue. Reading directly will ignore the write hence data is missing.

Reply:
> Stale read ts0 will ignore all locks because it's expected all locks will either be rollback or committed using a timestamp ts1 > ts0
It's problematic if the previous protocol is used in which the locks would be ignored.

We are currently considering changing the protocol to disallow ignoring locks even in cases of stale reads. As the assumptions are different, so the `max_ts` is also updated by the stale read requests in this PR.